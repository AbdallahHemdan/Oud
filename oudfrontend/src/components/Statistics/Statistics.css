.statistics-header {
  border-bottom: 1px solid #424242;
  font-size: 2rem;
  margin-left: 10px;
  margin-bottom: 20px;
}

.item-name {
  margin-left: 10px;
}

.statistics-data {
  margin: 50px 0px 50px 10px;
}

.stat-number {
  color: #f9b835;
}

.stat-card {
  min-width: 300px;
}
