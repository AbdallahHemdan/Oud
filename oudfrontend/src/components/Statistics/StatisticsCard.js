import React, { Component } from 'react'
import CountUp from "react-countup";
import { subUrl } from "./../../config/environment"

/**
 * @author AbdallahHemdan
 * @component
 * @description Statistics card is component to render the statistics card with all its statistics
 */
class StatisticsCard extends Component {

  /**
   * @function
   * @name render
   * @description Render Statistics cars
   * @returns {JSX} Component for App
   */

  render() {
    let {
      /**
       * @type {String}
       * #@description name of the song or the album of the current artist
       */
      name,
      /**
       * @type {String}
       * #@description image of the song or the album of the current artist
       */
      img,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album
       */
      listeners,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album
       */
      likes,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per day
       */
      listenersPerDay,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per week
       */
      listenersPerWeek,
      /**
       * @type {Integer}
       * #@description number of listeners of the current song or the current album per month
       */
      listenersPerMonth,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per day
       */
      likesPerDay,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per week
       */
      likesPerWeek,
      /**
       * @type {Integer}
       * #@description number of likes of the current song or the current album per month
       */
      likesPerMonth
    } = this.props.data;
    return (
      <>
        <h5 className="item-name"
          data-testid="item-name"
        >{name}</h5>
        <img src={img} alt="song" className="card"
          data-testid="item-image"
        />
        <div className="statistics-data"
          data-testid="statistics-data-container">
          <h5 data-testid="listeners">Listeners : <span className="stat-number">{listeners}</span></h5>
          <h5 data-testid="likes">Likes : <span className="stat-number">{likes}</span></h5>
          <h5 data-testid="listeners-per-day">Listeners per day : <span className="stat-number">{listenersPerDay}</span></h5>
          <h5 data-testid="listeners-per-week">Listeners per week : <span className="stat-number">{listenersPerWeek}</span></h5>
          <h5 data-testid="listeners-per-month">Listeners per month : <span className="stat-number">{listenersPerMonth}</span></h5>
          <h5 data-testid="likes-per-day">Likes per day : <span className="stat-number">{likesPerDay}</span></h5>
          <h5 data-testid="likes-per-week">Likes per week : <span className="stat-number">{likesPerWeek}</span></h5>
          <h5 data-testid="likes-per-month">Likes per month : <span className="stat-number">{likesPerMonth}</span></h5>
        </div>
      </>
    )
  }
}

export default StatisticsCard
