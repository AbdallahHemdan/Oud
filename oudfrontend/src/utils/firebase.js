import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
