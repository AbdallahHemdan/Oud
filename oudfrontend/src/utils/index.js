import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
