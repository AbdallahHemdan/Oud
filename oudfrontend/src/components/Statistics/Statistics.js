import React, { Component } from 'react'
import StatisticsCard from './StatisticsCard';
import axios from 'axios';
import { base, mockUrl } from "./../../config/environment";
import "./Statistics.css";


/**
 * @type {Array}
 * @description Array of the songs of the current artist
 */
const songs = [
  {
    "img": "https://i.pinimg.com/736x/00/a1/3c/00a13cf897548091f4042cba761ef00d--cd-cover-dance-music.jpg",
    "name": "Mail",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  },
  {
    "img": "https://i.pinimg.com/736x/cf/74/03/cf74032199d378808402329b3765ac72--music-albums-in-color.jpg",
    "name": "Segue o Baile",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  },
  {
    "img": "https://media1.popsugar-assets.com/files/thumbor/L-VT9k0GKckWWnpo3n2YBq4f9tE/fit-in/1024x1024/filters:format_auto-!!-:strip_icc-!!-/2017/08/03/846/n/37139775/2d0d134e5983773fe90560.65157921_edit_img_image_17921777_1501786115/i/Camila-Cabello-Havana-Song.jpg",
    "name": "Arai songs",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  },
  {
    "img": "https://i.pinimg.com/736x/ed/b1/10/edb1100257bb6d0ceee624696d9c9f1f--cover-art-album-covers.jpg",
    "name": "Rap songs",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  },
  {
    "img": "https://i.pinimg.com/736x/02/b8/94/02b894f7ea6ad9f724648ee511ad018f--edm-music-house-music.jpg",
    "name": "jazz songs",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  },
  {
    "img": "https://is4-ssl.mzstatic.com/image/thumb/Music7/v4/4b/06/29/4b062955-ecce-e362-78f1-025f18eed20a/source/1200x1200bb.jpg",
    "name": "Haly songs",
    "listeners": 3500,
    "likes": 1200,
    "listenersPerDay": 100,
    "listenersPerWeek": 750,
    "listenersPerMonth": 1230,
    "likesPerDay": 17,
    "likesPerWeek": 23,
    "likesPerMonth": 515
  }
];

/**
 * @author AbdallahHemdan
 * @component
 * @description Component to render the statistics of of the current artist songs and albums
 */
class Statistics extends Component {
  /**
   * @function
   * @name render
   * @description Render all the statistics of the current artist albums and songs
   *
   * @returns {JSX} Component for Home
  */
  render() {
    return (
      <>
        <div
          className="song-statistics"
          data-testid="statistics-songs"
        >
          <h1
            className="statistics-header"
            data-testid="statistics-header"
          >Songs</h1>
          <section
            className="statistics-cards row"
            data-testid="statistics-cards"
          >
            <div
              className="col-2 stat-card"
              data-testid="statistics-card"
            >
              <StatisticsCard data={songs[0]} />
            </div>
            <div
              className="col-2 stat-card"
              data-testid="statistics-card">
              <StatisticsCard data={songs[1]} />
            </div>
            <div
              className="col-2 stat-card"
              data-testid="statistics-card">
              <StatisticsCard data={songs[2]} />
            </div>
          </section>
        </div >
        <div className="album-statistics"
          data-testid="statistics-album">
          <h1
            className="statistics-header"
            data-testid="statistics-header">
            Albums
          </h1>
          <section
            className="statistics-cards row"
            data-testid="statistics-cards">
            <div
              className="col-2 stat-card"
              data-testid="statistics-card">
              <StatisticsCard data={songs[0]} />
            </div>
            <div
              className="col-2 stat-card"
              data-testid="statistics-card">
              <StatisticsCard data={songs[1]} />
            </div>
            <div
              className="col-2 stat-card"
              data-testid="statistics-card">
              <StatisticsCard data={songs[2]} />
            </div>
          </section>
        </div>
      </>
    )
  }
}

export default Statistics
