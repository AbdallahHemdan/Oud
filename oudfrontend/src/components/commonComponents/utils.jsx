import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}