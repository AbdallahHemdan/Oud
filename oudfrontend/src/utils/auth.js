import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
