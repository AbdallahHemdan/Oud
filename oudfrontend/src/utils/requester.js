import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
