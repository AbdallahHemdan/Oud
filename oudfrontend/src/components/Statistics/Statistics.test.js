import React from "react";
import { shallow } from 'enzyme';
import { findByTestAttr } from "./../../utils/index"
import Statistics from './Statistics';
import renderer from 'react-test-renderer';


const setup = (props = {}) => {
  const component = shallow(
    <Statistics {...props} />
  )
  return component;
};

describe('Statistics page Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  })

  it('Should render Loading Snipper Wrapper', () => {
    const wrapper = findByTestAttr(component, "statistics-songs");
    expect(wrapper.length).toBe(1);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "statistics-header");
    expect(wrapper.length).toBe(2);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "statistics-cards");
    expect(wrapper.length).toBe(2);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "statistics-card");
    expect(wrapper.length).toBe(6);
  });
  it('Should render Loading Snipper first Column', () => {
    const wrapper = findByTestAttr(component, "statistics-album");
    expect(wrapper.length).toBe(1);
  });

  describe('snapshot test', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<Statistics />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});