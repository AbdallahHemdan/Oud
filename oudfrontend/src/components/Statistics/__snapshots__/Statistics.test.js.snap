// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics page Component snapshot test renders correctly 1`] = `
Array [
  <div
    className="song-statistics"
    data-testid="statistics-songs"
  >
    <h1
      className="statistics-header"
      data-testid="statistics-header"
    >
      Songs
    </h1>
    <section
      className="statistics-cards row"
      data-testid="statistics-cards"
    >
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Mail
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://i.pinimg.com/736x/00/a1/3c/00a13cf897548091f4042cba761ef00d--cd-cover-dance-music.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Segue o Baile
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://i.pinimg.com/736x/cf/74/03/cf74032199d378808402329b3765ac72--music-albums-in-color.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Arai songs
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://media1.popsugar-assets.com/files/thumbor/L-VT9k0GKckWWnpo3n2YBq4f9tE/fit-in/1024x1024/filters:format_auto-!!-:strip_icc-!!-/2017/08/03/846/n/37139775/2d0d134e5983773fe90560.65157921_edit_img_image_17921777_1501786115/i/Camila-Cabello-Havana-Song.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
    </section>
  </div>,
  <div
    className="album-statistics"
    data-testid="statistics-album"
  >
    <h1
      className="statistics-header"
      data-testid="statistics-header"
    >
      Albums
    </h1>
    <section
      className="statistics-cards row"
      data-testid="statistics-cards"
    >
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Mail
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://i.pinimg.com/736x/00/a1/3c/00a13cf897548091f4042cba761ef00d--cd-cover-dance-music.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Segue o Baile
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://i.pinimg.com/736x/cf/74/03/cf74032199d378808402329b3765ac72--music-albums-in-color.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
      <div
        className="col-2 stat-card"
        data-testid="statistics-card"
      >
        <h5
          className="item-name"
          data-testid="item-name"
        >
          Arai songs
        </h5>
        <img
          alt="song"
          className="card"
          data-testid="item-image"
          src="https://media1.popsugar-assets.com/files/thumbor/L-VT9k0GKckWWnpo3n2YBq4f9tE/fit-in/1024x1024/filters:format_auto-!!-:strip_icc-!!-/2017/08/03/846/n/37139775/2d0d134e5983773fe90560.65157921_edit_img_image_17921777_1501786115/i/Camila-Cabello-Havana-Song.jpg"
        />
        <div
          className="statistics-data"
          data-testid="statistics-data-container"
        >
          <h5
            data-testid="listeners"
          >
            Listeners : 
            <span
              className="stat-number"
            >
              3500
            </span>
          </h5>
          <h5
            data-testid="likes"
          >
            Likes : 
            <span
              className="stat-number"
            >
              1200
            </span>
          </h5>
          <h5
            data-testid="listeners-per-day"
          >
            Listeners per day : 
            <span
              className="stat-number"
            >
              100
            </span>
          </h5>
          <h5
            data-testid="listeners-per-week"
          >
            Listeners per week : 
            <span
              className="stat-number"
            >
              750
            </span>
          </h5>
          <h5
            data-testid="listeners-per-month"
          >
            Listeners per month : 
            <span
              className="stat-number"
            >
              1230
            </span>
          </h5>
          <h5
            data-testid="likes-per-day"
          >
            Likes per day : 
            <span
              className="stat-number"
            >
              17
            </span>
          </h5>
          <h5
            data-testid="likes-per-week"
          >
            Likes per week : 
            <span
              className="stat-number"
            >
              23
            </span>
          </h5>
          <h5
            data-testid="likes-per-month"
          >
            Likes per month : 
            <span
              className="stat-number"
            >
              515
            </span>
          </h5>
        </div>
      </div>
    </section>
  </div>,
]
`;
